api/v1alpha1/foundationloadbalancerconfig_types.go

```diff
@@ -0,0 +1,229 @@
+	StoragePolicy string `json:"storagePolicy"`
+
+	// Version number desired by the operator.
+	//
+	// Defaults to the latest available.
+	// +optional
+	Version string `json:"version,omitempty"`
+
+	// Zones contains the names of zones eligible for placing nodes. Zones must be one of the
+	// AvailabilityZones defined and eligible for placement on the cluster.
+	//
+	// If no zones are provided, then you must provide a PlacementSpec.
+	// +optional
+	Zones []string `json:"zones,omitempty"`
+
+	// AvailabilityMode defines how the availability of the solution is deployed and configured.
+	// +kubebuilder:validation:Enum=active-passive;single-node
+	// +kubebuilder:default=active-passive
+	AvailabilityMode FoundationLoadBalancerAvailabilityMode `json:"availabilityMode"`
+
+	// ActivePassiveAvailabilityMode configures the load balancer in active-passive configuration.
+	// Active-passive configuration consists of a two node deployment with one node configured to
+	// actively service traffic with the second node in standby mode. When the service detects the
+	// active node is unhealthy, traffic will be moved to the passive node after a short delay.
+	// Connections may be dropped on fail-over.
+	//
+	// +optional
+	ActivePassiveAvailabilityMode *ActivePassiveAvailabilityMode `json:"activePassiveSpec,omitempty"`
+
+	// SingleNodeAvailabilityMode deploys a single node to serve load balancer traffic. If the node
+	// fails, the service will attempt to redeploy it, but redeployment is best-effort and depends on
+	// the health of the underlying infrastructure. You must select
+	//
+	// +optional
+	SingleNodeAvailabilityMode *SingleModeAvailabilityMode `json:"singleNodeSpec,omitempty"`
+
+	// PlacementSpec is optional configuration defining custom placement of load balancer nodes.
+	//
+	// If Zones are specified, this field is ignored.
+	// If Zones are not specified, this field must be set.
+	// +optional
+	PlacementSpec []CustomPlacementSpec `json:"placementSpec,omitempty"`
+}
+
+// ActivePassiveAvailabilityMode deploys two nodes in Active-Passive mode where one node is set into
+// active state and is responsible for serving traffic, and one node is passive -
+// awaiting a fail-over event. When a fail-over occurs, connections to and from the load balancer
+// may be reset.
+type ActivePassiveAvailabilityMode struct {
+	// Replicas describes the total number of deployed nodes. Defaults to 2.
+	//
+	// +kubebuilder:validation:Maximum=2
+	// +optional
+	Replicas *uint32 `json:"replicas,omitempty"`
+}
+
+// SingleModeAvailabilityMode defines single node configuration. Single node configuration involves
+// trading availability in return for reduced resource consumption. Upon node failure, redeployment will
+// be attempted on a best-effort basis.
+type SingleModeAvailabilityMode struct {
+	// Replicas describes the total number of deployed nodes. Defaults to 1.
+	//
+	// +kubebuilder:validation:Maximum=1
```

Comment:
Do you want a ``//+kubebuilder:default=1' here? I'm not totally sure that this field needs to be a pointer.